src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,183 @@
+    };
+    initialConstraint: HybridConstraint | string;
+    update: (obj: mendix.lib.MxObject | null, callback?: () => void) => void;
+    _entity: string;
+}
+
+export interface ContainerState {
+    listViewAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class CheckboxFilterContainer extends Component<ContainerProps, ContainerState> {
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listViewAvailable: true };
+        this.applyFilter = this.applyFilter.bind(this);
+        // Ensures that the listView is connected so the widget doesn't break in mobile due to unpredictable render time
+        dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-checkbox-filter", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            this.renderAlert(),
+            this.renderComponent()
+        );
+    }
+
+    componentWillReceiveProps(nextProps: ContainerProps) {
+        this.applyFilter(nextProps.defaultChecked);
+    }
+
+    private renderAlert() {
+        const message = Utils.validate({
+            ...this.props as ContainerProps,
+            filterNode: this.state.targetNode,
+            targetListView: this.state.targetListView
+        });
+
+        return createElement(Alert, {
+            className: "widget-checkbox-filter-alert",
+            message
+        });
+    }
+
+    private renderComponent(): ReactElement<CheckboxFilterProps> {
+        if (this.state.validationPassed) {
+
+            return createElement(CheckboxFilter, {
+                handleChange: this.applyFilter,
+                isChecked: this.props.defaultChecked
+            });
+        }
+
+        return null;
+    }
+
+    private applyFilter(isChecked: boolean) {
+        const { targetListView, targetNode } = this.state;
+        // To support multiple filters. We attach each checkbox-filter-widget's selected constraint
+        // On the listView's custom 'filter' object.
+        if (targetListView && targetListView._datasource) {
+            this.showLoader(targetNode);
+            const attribute = isChecked ? this.props.attribute : this.props.unCheckedAttribute;
+            const filterBy = isChecked ? this.props.filterBy : this.props.unCheckedFilterBy;
+            const constraint = isChecked ? this.props.constraint : this.props.unCheckedConstraint;
+            const attributeValue = isChecked ? this.props.attributeValue : this.props.unCheckedAttributeValue;
+
+            if (filterBy === "XPath") {
+                targetListView.filter[this.props.friendlyId] = constraint.indexOf(`[%CurrentObject%]'`) !== -1
+                    ? constraint.replace(`'[%CurrentObject%]'`, this.props.mxObject.getGuid())
```

Comment:
If there is no context in around the widget, will `mxObject` be undefined causing this line toe break?